scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,413 @@
+    return address.isTextual()
+        ? new RevioPacBioProcessor(builder, address.textValue().replaceAll("/+$", ""))
+        : null;
+  }
+
+  @Override
+  public Stream<File> getRunsFromRoot(File root) {
+    return Arrays.stream(
+        Objects.requireNonNull(
+            root.listFiles(f -> f.isDirectory() && RUN_DIRECTORY.matcher(f.getName()).matches())));
+  }
+
+  @Override
+  public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
+    // We create one DTO for a run, but there are going to be many wells with
+    // independent and
+    // duplicate metadata that will simply
+    // overwrite in the shared DTO. If the data differs, the last well wins.
+    PacBioNotificationDto dto = new PacBioNotificationDto();
+    dto.setPairedEndRun(false);
+    dto.setSequencerFolderPath(runDirectory.getAbsolutePath());
+
+    // Get the number of SMRT Cells in the run directory
+    int smrtCellCount =
+        (int)
+            Arrays.stream(runDirectory.listFiles())
+                .filter(
+                    cellDirectory ->
+                        cellDirectory.isDirectory()
+                            && REVIO_CELL_DIRECTORY.test(cellDirectory.getName()))
+                .count();
+    dto.setLaneCount(smrtCellCount);
+
+    // Check for presence of Transfer_Test file initially to grab start time
+    try (Stream<Path> stream = Files.walk(runDirectory.toPath())) {
+      List<Path> transferTestFiles =
+          stream
+              .filter(Files::isRegularFile)
+              .filter(file -> TRANSFER_TEST.test(String.valueOf(file.getFileName())))
+              .toList();
+
+      // Need to take the list of file paths and get the earliest creation time
+      Optional<Path> earliestCreatedFilePath =
+          transferTestFiles
+              .stream()
+              .min(Comparator.comparing(filePath -> getFileCreationTime(filePath.toFile())));
+      dto.setStartDate(
+          getFileCreationTime(
+              Objects.requireNonNull(earliestCreatedFilePath.orElse(null)).toFile()));
+    } catch (IOException e) {
+      log.warn("Transfer_Test file not present");
+    }
+
+    // Grab the .metadata.xml and begin processing
+    getSubDirectory(runDirectory, "metadata")
+        .flatMap(
+            metadataDirectory ->
+                Stream.of(Objects.requireNonNull(metadataDirectory.listFiles()))
+                    .filter(file -> file.getName().endsWith(".metadata.xml")))
+        .map(RunProcessor::parseXml)
+        .filter(Optional::isPresent)
+        .forEach(metadata -> processMetadata(metadata.get(), dto, tz));
+
+    // Check for .transferdone and Transfer_Test in all SMRT Cells
+    // to consider the run complete
+    if (isRunComplete(runDirectory, smrtCellCount)) {
+      dto.setHealthType(HealthType.COMPLETED);
+
+      /*
+       We have all the expected files. Initially use file creation time, but if
+       there is a timestamp from pbereports.log, use that instead
```

Comment:
Why did you do it in this order instead of checking for pbereports.log first, then file creation time as a backup?